### PR TITLE
possible solution to issue #69

### LINKED ISSUE #69: Initialize ontology continuous integration launcher

### PR DESCRIPTION
**Pull request by othub**

### CODE REVIEW

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 1]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
-  private final val ONTOLOGY_FILES      =   List[Source]()
-  private final val INSTANCE_FILES      =   List[Source]()
-  private final val SHAPE_EX_FILES      =   List[Source]()
-  private final val SHAPE_MAP_FILES     =   List[Source]()
-  private final val RES_SHAPE_MAP_FILES =   List[Source]()
+  private final var ONTOLOGY_FILES      =   List[Source]()
+  private final var INSTANCE_FILES      =   List[Source]()
+  private final var SHAPE_EX_FILES      =   List[Source]()
+  private final var SHAPE_MAP_FILES     =   List[Source]()
+  private final var RES_SHAPE_MAP_FILES =   List[Source]()
```
Comment (thewilly) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Please do not change value definitions to variables, you just neet to add elements to the already created lists.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 2]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
 
   /**
-   * TO DO...
+   * adding files to list from ../src
    */
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Please add comments explaining what you are doing.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala (hunk 1 above)
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Please do not change value definitions to variables, you just neet to add elements to the already created lists.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 3]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
   def populateOntologyFiles(): Unit = {
-    // TO DO...
+    val files = getListOfFiles("../src", ".ttl")
+    var list = List[Source]()
+    for (dir <- files){
+      println(dir)
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Logging should not be done through console prints. If you want to add logging please use an apporpiate framework. Else you can just remove it.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 4]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
+  private final var RES_SHAPE_MAP_FILES =   List[Source]()
 
   /**
-   * TO DO...
+   * adding files to list from ../src
    */
   def populateOntologyFiles(): Unit = {
-    // TO DO...
+    val files = getListOfFiles("../src", ".ttl")
+    var list = List[Source]()
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: You dont neet to use a temporal list. Already have a dedicated one defined as a value previously.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 5]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
+    for (dir <- files){
+      println(dir)
+      list=list:+Source.fromFile(dir)
+    }
+    ONTOLOGY_FILES = list
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: This should be not done, ONTOLOGY_FILES should be a value and therefore should not be reassigned.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 6]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
-   * TO DO...
+   * adding files to list from ../src
    */
   def populateOntologyFiles(): Unit = {
-    // TO DO...
+    val files = getListOfFiles("../src", ".ttl")
+    var list = List[Source]()
+    for (dir <- files){
+      println(dir)
+      list=list:+Source.fromFile(dir)
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: You can do:
```scala
ONTOLOGY_FILES.appended(dir)
```

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 7]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
 
   /**
-   * TO DO...
+   * adding files to list from ../src
    */
   def populateOntologyFiles(): Unit = {
-    // TO DO...
+    val files = getListOfFiles("../src", ".ttl")
+    var list = List[Source]()
+    for (dir <- files){
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Use variable names with meaning, if you are iterating over a list of files the iterator should be name file.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 8]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
+      println(dir)
+      list=list:+Source.fromFile(dir)
+    }
+    ONTOLOGY_FILES = list
   }
 
   /**
-   *
+   * adding files from /tests with .ttl extension
    */
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Please describe what you are doing in the cooments.

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala [hunk 9]
```diff
@@ -34,45 +34,85 @@ import scala.io.Source
...
+
+  /**
+   * return list of paths of files in a specific directory and a specific extension
+   */
+  def getListOfFiles(dir: String, ext: String):List[String] = {
+    val d = new File(dir)
+    d.listFiles.
+    filter { f => f.isFile && (f.getName.endsWith(ext)) }.
+    map(_.getPath).toList
   }
```
Comment (thewillyhuman) [inline ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: - For new functions that will be only called from this class define them as private please.
- Also use `@return` and good documentation practices. 
- Local values / variables should containg a name that represents its content. If its a directory then a good name could be `directory` or `dir` if already in use maybe some variable / value / parameter is not well named.

**Review by thewillyhuman — CHANGES_REQUESTED** [review]
Please read all the comments and proceed with the corresponding changes. Some comments can be applied to more than one method.

If any doubts please feel free to ask.

Thanks!!

**Comment by othub** [comment]
@thewilly i tried not changing them and keeping the **val** but apparently it shows an error as you can't add en element to a constant or change it and that's why i changed it to **var**

File: ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala (hunk 6 above)
Comment (thewillyhuman) [reply ci/src/test/scala/es/weso/asio/ci/ContinuousIntegrationLauncher.scala]: Here is an small example that works.

```scala
  def populateOntologyFiles(): Unit = {
    val files = getListOfFiles("../src", ".ttl")
    for ( file<- files) {
      ONTOLOGY_FILES appended Source.fromFile(file)
    }
  }
```

**Comment by othub** [comment]
i updated the code with the changes from the review, let me know it's okay

**Review by thewillyhuman — APPROVED** [review]

**Pull request merged**